llama_stack/providers/inline/post_training/torchtune/post_training.py

```diff
@@ -111,19 +123,54 @@ async def preference_optimize(
     ) -> PostTrainingJob: ...
 
     async def get_training_jobs(self) -> ListPostTrainingJobsResponse:
-        return ListPostTrainingJobsResponse(data=[PostTrainingJob(job_uuid=uuid_) for uuid_ in self.jobs])
+        return ListPostTrainingJobsResponse(
+            data=[PostTrainingJob(job_uuid=job.id) for job in self._scheduler.get_jobs()]
+        )
+
+    @staticmethod
+    def _get_artifact_metadata_by_type(job, artifact_type):
+        return [artifact.metadata for artifact in job.artifacts if artifact.type == artifact_type]
+
+    @classmethod
+    def _get_checkpoints(cls, job):
+        return cls._get_artifact_metadata_by_type(job, _ARTIFACT_TYPE_CHECKPOINT)
+
+    @classmethod
+    def _get_resources_allocated(cls, job):
+        data = cls._get_artifact_metadata_by_type(job, _ARTIFACT_TYPE_RESOURCES_STATS)
+        return data[0] if data else None
 
     @webmethod(route="/post-training/job/status")
     async def get_training_job_status(self, job_uuid: str) -> Optional[PostTrainingJobStatusResponse]:
-        return self.jobs.get(job_uuid, None)
+        job = self._scheduler.get_job(job_uuid)
+
+        match job.status:
+            case SchedulerJobStatus.new | SchedulerJobStatus.scheduled:
+                status = JobStatus.scheduled
```

Comment:
any specific reason for not differentiating between new and scheduled here?

Reply:
Currently, JobStatus (on API) doesn't have "new" as an option. I will add a TODO here to revisit it.